In debug mode, don't enable the linker by default

SteveSandersonMS opened the pull request:
This makes the inner-loop dev experience much better. We're also clearer these days about logging a warning about the effect on app size and the need to test published/release builds regularly.

SteveSandersonMS committed:
In debug mode, don't enable the linker by default

pranavkm committed:
Fixup

pranavkm commented on src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets:
```diff
@@ -66,7 +66,7 @@
   <Target Name="_ResolveBlazorOutputs" DependsOnTargets="_ResolveBlazorOutputsWhenLinked;_ResolveBlazorOutputsWhenNotLinked">
     <Error
-      Message="Unrecongnized value for BlazorWebAssemblyEnableLinking: '$(BlazorWebAssemblyEnableLinking)'. Valid values are 'true' or 'false'."
+      Text="Unrecongnized value for BlazorWebAssemblyEnableLinking: '$(BlazorWebAssemblyEnableLinking)'. Valid values are 'true' or 'false'."
```
Sorry, my bad. I thought I tested this

pranavkm committed:
Update Blazor.MonoRuntime.targets

SteveSandersonMS committed:
Ensure we have a true/false value. Stop inferring from BlazorLinkOnBuild.

SteveSandersonMS committed:
Avoid doing work for ServiceWorkerAssetsManifest when it's not being used

SteveSandersonMS committed:
React to BlazorLinkOnBuild->BlazorWebAssemblyEnableLinking rename

SteveSandersonMS commented:
This PR has expanded in scope a bit now:

 * Also made a fix to ServiceWorkerAssetsManifest target to avoid performing work when we're not using it anyway
 * Fixed some vestigial uses of the property name `BlazorLinkOnBuild`
 * Changed to be more aggressive about forcing people to update from `BlazorLinkOnBuild` to `BlazorWebAssemblyEnableLinking` by failing their builds otherwise (because if we don't, it's harder to break people later)

javiercn reviewed (approved):
Reviewed the update. Changes look good.
